File: fixtures/scale/kotlin/large/src/cryptobench/store/ResultStore.kt
package cryptobench.store

import cryptobench.core.SuiteResult
import java.io.BufferedWriter
import java.nio.file.Files
import java.nio.file.Path

/** Appends one JSON line per finished suite to results/run.jsonl. */
class ResultStore private constructor(private val writer: BufferedWriter) {

    fun append(result: SuiteResult) {
        writer.write(toJsonLine(ResultRecord.of(result)))
        writer.newLine()
        writer.flush()
    }

    fun close() {
        writer.close()
    }

    companion object {
        fun openAt(directory: Path): ResultStore {
            Files.createDirectories(directory)
            return ResultStore(Files.newBufferedWriter(directory.resolve("run.jsonl")))
        }
    }
}

File: fixtures/scale/dart/large/lib/store/result_store.dart
import 'dart:io';

import '../core/suite_result.dart';
import 'json_lines.dart';
import 'result_record.dart';

/// Appends one JSON line per finished suite to results/run.jsonl.
class ResultStore {
  final IOSink _sink;

  ResultStore._(this._sink);

  factory ResultStore.openAt(String directory) {
    final dir = Directory(directory)..createSync(recursive: true);
    final file = File('${dir.path}/run.jsonl');
    return ResultStore._(file.openWrite());
  }

  void append(SuiteResult result) {
    _sink.writeln(toJsonLine(ResultRecord.of(result)));
  }

  void close() {
    _sink.close();
  }
}
